### ISSUE TITLE
Nil error on failure to authenticate

### ISSUE THREAD

**Issue by vinyar** [pendrica/kitchen-azurerm#5]
If credentials file is set up incorrectly, `kitchen create` fails with `[undefined method `[]' for nil:NilClass]`

The real error as can be seen from pry is failure to authenticate

```
$ ~/cookbooks/kitchen_azurerm $ kitchen create
-----> Starting Kitchen (v1.4.2)
-----> Creating <default-windows2012-r2>...

From: /Users/alexvinyar/.chefdk/gem/ruby/2.1.0/gems/kitchen-azurerm-0.2.1/lib/kitchen/driver/azurerm.rb @ line 61 Kitchen::Driver::Azurerm#create:

    56:         @resource_management_client.subscription_id = config[:subscription_id]
    57:
    58:         # Create Resource Group
    59:         resource_group = ::Azure::ARM::Resources::Models::ResourceGroup.new
    60:         resource_group.location = config[:location]
 => 61:         require 'pry';binding.pry
    62:         begin
    63:           info "Creating Resource Group: #{state[:azure_resource_group_name]}"
    64:           resource_management_client.resource_groups.create_or_update(state[:azure_resource_group_name], resource_group).value!
    65:         rescue ::MsRestAzure::AzureOperationError => operation_error
    66:           info operation_error.body['error']
```
 Hitting the method manually <
```
2.1.6 (#<Kitchen::Driver::Azurerm:0x007fb11da5b470>):0 > 
resource_management_client.resource_groups.create_or_update(state[:azure_resource_group_name], resource_group).value!
MsRestAzure::AzureOperationError: Couldn't login to Azure, please verify your tenant id, client id and client secret
from /Users/alexvinyar/.chefdk/gem/ruby/2.1.0/gems/ms_rest_azure-0.1.1/lib/ms_rest_azure/credentials/application_token_provider.rb:108:in `acquire_token'
2.1.6 (#<Kitchen::Driver::Azurerm:0x007fb11da5b470>):0 > exit
```
The error kitchen presents (even in debug mode)
```
       Creating Resource Group: kitchen-default-windows2012-r2-20151027T052831
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #create action: [undefined method `[]' for nil:NilClass]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

**Comment by stuartpreston** [comment]
Looks like the SDK made some changes that means the error is not available the same way. I'll have a look, thanks.

**Comment by vinyar** [comment]
:+1:

**Comment by stuartpreston** [comment]
Just retested this and the lack of error handling is still an issue.

**Comment by stuartpreston** [comment]
This has been fixed in 0.3.0-pre onwards.

**Comment by stuartpreston** [comment]
Please reopen if the issue remains
